frontends/pytorch/csrc/builder/graph_importer.h

```diff
@@ -0,0 +1,117 @@
+//===- graph_importer.h -----------------------------------------*- C++ -*-===//
+//
+// This file is licensed under a pytorch-style license
+// See frontends/pytorch/LICENSE for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef NPCOMP_FRONTENDS_PYTORCH_CSRC_GRAPH_IMPORTER_H
+#define NPCOMP_FRONTENDS_PYTORCH_CSRC_GRAPH_IMPORTER_H
+
+#include <memory>
+
+#include "../pybind.h"
+#include "func_builder.h"
+
+#include "mlir-c/IR.h"
+
+#include <torch/csrc/jit/api/compilation_unit.h>
+#include <torch/csrc/jit/ir/ir.h>
+
+namespace torch_mlir {
+
+/// Main entry-point for importing graphs.
+/// In torch terminology, a Graph is a function. We materialize it as at
+/// least one MLIR function, but may specialize multiple versions of it,
+/// depending on usage.
+///
+/// Since graph functions typically have enough annotations for the most
+/// generic form of every type (i.e. Tensor, List, etc), and since we often
+/// want to multi-version specific specializations, we take the approach of
+/// generating a '$generic' suffixed function at that level and then generate
+/// the actual named function with using a 'numpy.generic_call' op to invoke
+/// the generic function with metadata controlling how it is legal to
+/// specialize. This leaves the process of inlining and expanding the
+/// specializations to compiler passes.
+class GraphImporter : public std::enable_shared_from_this<GraphImporter> {
+public:
+  struct MlirMappingOptions {
```

Comment:
some comments on MlirMappingOptions? I see at least that there are some shenanigans around `""` being some sort of default/auto thing?